quickwit/quickwit-control-plane/src/model/shard_table.rs

```diff
@@ -146,6 +172,20 @@ fn remove_shard_from_ingesters_internal(
 }
 
 impl ShardTable {
+    /// Returns a ShardLocations object that maps each shard to the list of indexers hosting it.
+    /// All shards are considered regardless of their state (including unavailable).
+    pub fn shard_locations(&self) -> ShardLocations {
```

Comment:
I think it's more natural to create the shard locations from iterating over the shards directly rather than iterating over the ingesters. It is because of the lifetime?

```python
for table_entry in self.shard_table {
  for shard in table_entry {
       shard_locations.add_location(shard.shard_id, shard.leader_id, shard.follower_id);
  }
}

Reply:
No good reason...
I thought it was more natural that way. 
A minor benefit is that I get reference to node ids maybe.